src/main/java/space/obminyashka/items_exchange/dao/AdvertisementRepository.java

```diff
@@ -30,6 +30,9 @@ public interface AdvertisementRepository extends JpaRepository<Advertisement, UU
     @Query("SELECT a FROM Advertisement a WHERE LOWER(a.topic) IN :topics")
     Page<Advertisement> search(@Param("topics") Set<String> topics, Pageable pageable);
 
+    @Query("SELECT a FROM Advertisement a WHERE :categoryId is null or a.subcategory.category.id = :categoryId")
+    Page<Advertisement> findAdvertisementByCategoryId(@Param("categoryId") Long categoryId, Pageable pageable);
```

Comment:
If I remember well you can omit Param annotation for the method's parameters with the same as the value in the brackets. Would you mind checking if that is correct?

Reply:
Without annotation does not work